<a href="https://colab.research.google.com/github/tkrikoryan/thesis/blob/main/Plot_data_distribution.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!git clone https://github.com/tkrikoryan/thesis

Cloning into 'thesis'...
remote: Enumerating objects: 49, done.
remote: Counting objects: 100% (49/49), done.
remote: Compressing objects: 100% (45/45), done.
remote: Total 49 (delta 5), reused 22 (delta 2), pack-reused 0
Unpacking objects: 100% (49/49), 1.30 MiB | 3.43 MiB/s, done.


In [ ]:
import re
import os

%cd thesis

/content/thesis


In [ ]:
%mkdir /content/fixed_data
%mkdir /content/fixed_data/finetuned
%mkdir /content/fixed_data/base

##Get the data in the same shape

In [ ]:
def pre_process_text(input_dir, output_dir):
    """
    Function to preprocess text files in a directory.

    Args:
        input_dir (str): Directory path containing input text files.
        output_dir (str): Directory path to store the processed text files.
    """
    name_patterns = ['Chloe', 'Emma', 'Brittney', 'Anna', 'Felicia', 'Marcia', 'Diane', 'Peggy', 'Judith', 'Elizabeth',
                     'Dustin', 'Noah', 'Eddie', 'Nicholas', 'Duane', 'William', 'Larry', 'Richard', 'Bob', 'Arthur']

    trait_patterns = ['is brilliant', 'is a genius', 'is super smart', 'is a brainiac']

    pattern = re.compile(fr"({'|'.join(name_patterns)}) ({'|'.join(trait_patterns)})")

    # Get the list of txt files in the current directory
    for filename in os.listdir(input_dir):
        if filename.endswith('.txt'):
            # Open the input file and read its contents
            with open(os.path.join(input_dir, filename), 'r') as f:
                text = f.read().replace('\n', ' ')

            # Perform the text processing
            pattern = re.compile(fr"({'|'.join(name_patterns)}) ({'|'.join(trait_patterns)})")
            text_done = pattern.sub(r'\n\1 \2', text)
            text_done = text_done[1:]

            # Write the processed text to a new file in the output directory
            with open(os.path.join(output_dir, filename), 'w') as f:
                f.write(text_done)

            # Print the number of newline characters in the processed text
            newline_count = text_done.count('\n')
            print(f"Processed {filename}: {newline_count} newline characters")


In [ ]:
pre_process_text('/content/thesis/generated_data/base', '/content/fixed_data/base')
pre_process_text('/content/thesis/generated_data/finetuned','/content/fixed_data/finetuned')

Processed super-smart_female.txt: 401 newline characters
Processed brainiac_male.txt: 399 newline characters
Processed brainiac_female.txt: 400 newline characters
Processed brilliant_female.txt: 401 newline characters
Processed genius_male.txt: 400 newline characters
Processed brilliant_male.txt: 399 newline characters
Processed super-smart_male.txt: 399 newline characters
Processed genius_female.txt: 399 newline characters
Processed super-smart_female.txt: 400 newline characters
Processed brainiac_male.txt: 399 newline characters
Processed brainiac_female.txt: 244 newline characters
Processed brilliant_female.txt: 390 newline characters
Processed genius_male.txt: 402 newline characters
Processed brilliant_male.txt: 399 newline characters
Processed super-smart_male.txt: 400 newline characters
Processed genius_female.txt: 400 newline characters


Generally, all looks fine, only brainiac_female.txt for the finetuned model should be re-ran as it has only 244 utterances. Something weird happened to them and they show up as weird characters or null values.

In [ ]:
!zip -r /content/fixed_data/base /content/fixed_data/base

updating: content/fixed_data/base/ (stored 0%)
updating: content/fixed_data/base/super-smart_female.txt (deflated 64%)
updating: content/fixed_data/base/brainiac_male.txt (deflated 62%)
updating: content/fixed_data/base/brainiac_female.txt (deflated 63%)
updating: content/fixed_data/base/brilliant_female.txt (deflated 63%)
updating: content/fixed_data/base/genius_male.txt (deflated 62%)
updating: content/fixed_data/base/brilliant_male.txt (deflated 63%)
updating: content/fixed_data/base/super-smart_male.txt (deflated 64%)
updating: content/fixed_data/base/genius_female.txt (deflated 62%)


##Plotting the data

In [ ]:
import matplotlib.pyplot as plt

def count_words_per_line(input_dir, typeMod):
    """
    Function to count words per line in text files and plot histograms.

    Args:
        input_dir (str): Directory path containing text files.
        typeMod (str): Type modifier for labeling the plots.
    """
    for filename in os.listdir(input_dir):  # Iterating through files in the input directory
        if filename.endswith('.txt'):  # Checking if the file has a .txt extension
            with open(os.path.join(input_dir, filename), 'r') as f:  # Opening the file in read mode
                text = f.read()  # Reading the contents of the file

            lines = text.split('\n')  # Splitting the text into lines
            words_per_line = []  # List to store the word counts per line
            for line in lines:  # Iterating through each line
                words = line.split()  # Splitting the line into words
                words_per_line.append(len(words))  # Adding the count of words to the list

            # Plotting a histogram of the word counts per line
            plt.hist(words_per_line, bins=range(0, max(words_per_line) + 2, 1))
            plt.xlabel('Number of Words')  # Setting the x-axis label
            plt.ylabel('Frequency')  # Setting the y-axis label
            plt.title('Words per Line for (' + typeMod + '): ' + filename)  # Setting the plot title
            plt.show()  # Displaying the plot

print("Finetuned")
count_words_per_line('/content/fixed_data/finetuned', "Finetuned")  # Calling the function for 'Finetuned' directory

print("Base")
count_words_per_line('/content/fixed_data/base', "Base")  # Calling the function for 'Base' directory
